pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -454,6 +456,14 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	// Return early and stop processing if Agent install strategy is in play. The controllers that
+	// handle this portion of the API currently live in the assisted service repo, rather than hive.
+	// This will hopefully change in the future.
+	if cd.Spec.Provisioning != nil && cd.Spec.Provisioning.InstallStrategy != nil && cd.Spec.Provisioning.InstallStrategy.Agent != nil {
+		cdLog.Info("skipping processing of agent install strategy cluster")
+		return reconcile.Result{}, nil
+	}
+
```

Comment:
for now this might be okay, but i think we will have to define a more appropriate boundary because this controller does a lot more than just provisioning...

Reply:
Yes I have a card roughly about this for next sprint, you are right, we need to think carefully about this and make sure that hive still does what it should for an agent based cluster.

I also want to discuss a little more about the plans to get the controllers doing this work into hive.

cc @filanov FYI